### PR TITLE
Add multiarch support for CIM

### PR DESCRIPTION
**Pull request by jtomasek**
Required by https://github.com/stolostron/console/pull/2264

### CODE REVIEW

File: src/cim/components/ClusterDeployment/ArmCheckbox.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+          name={name}
+          isDisabled={featureSupportLevelContext.isFeatureDisabled(
+            openshiftVersion,
+            'ARM64_ARCHITECTURE',
+          )}
+          label={<Label version={openshiftVersion} />}
+          aria-describedby={`${fieldId}-helper`}
+          description={
+            <HelperText fieldId={fieldId}>
+              Make sure all the hosts are using arm64 CPU architecture.
```
Comment (mareklibra) [inline src/cim/components/ClusterDeployment/ArmCheckbox.tsx:90]: missing translation

File: locales/en/translation.json [hunk 2]
```diff
@@ -678,9 +684,11 @@
...
   "ai:You have opted out of formatting bootable disks on some hosts. To ensure the hosts reboot into the expected installation disk, manual user intervention may be required during OpenShift installation.": "You have opted out of formatting bootable disks on some hosts. To ensure the hosts reboot into the expected installation disk, manual user intervention may be required during OpenShift installation.",
   "ai:Your libvirt virtual machines should be configured to restart automatically after a reboot. You can check this by running:": "Your libvirt virtual machines should be configured to restart automatically after a reboot. You can check this by running:",
-  "ai:Your own NTP (Network Time Protocol) sources": "Your own NTP (Network Time Protocol) sources"
+  "ai:Your own NTP (Network Time Protocol) sources": "Your own NTP (Network Time Protocol) sources",
+  "Displaying only hosts with {{cpuArchitecture}} architecture in the table.": "Displaying only hosts with {{cpuArchitecture}} architecture in the table."
```
Comment (mareklibra) [inline locales/en/translation.json:693]: missing `ai:` namespace

File: src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx [hunk 3]
```diff
@@ -66,7 +71,15 @@ const ClusterDeploymentHostsSelectionAdvanced = <T extends FormValues>({
             labelKeysFilter={infraEnvLabelKeys}
           />
         </GridItem>
-
+        {cpuArchitecture && (
+          <Alert
+            variant={AlertVariant.info}
+            title={t('Displaying only hosts with {{cpuArchitecture}} architecture in the table.', {
```
Comment (mareklibra) [inline src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx:77]: `ai:`

File: src/cim/components/featureSupportLevels/featureStateUtils.tsx [hunk 4]
```diff
@@ -1,28 +1,53 @@
...
+  isSupported: boolean,
+  t: TFunction,
+) => {
+  if (clusterExists) {
+    return getClusterExistsReason(t);
+  }
+  if (!isSupported) {
+    return t('ai:arm64 is not supported in this OpenShift version');
+  }
+  // TODO(jtomasek): versionOptions derived from clusterimagesets need to include cpuArchitectures array to correctly determine this
```
Comment (mareklibra) [inline src/cim/components/featureSupportLevels/featureStateUtils.tsx:30]: Is that blocked on something recently?

File: src/common/components/hosts/tableUtils.tsx [hunk 5]
```diff
@@ -178,6 +178,24 @@ export const discoveredAtColumn: TableRow<Host> = {
   },
 };
 
+export const cpuArchitectureColumn: TableRow<Host> = {
+  header: {
+    title: 'CPU Architecture',
```
Comment (mareklibra) [inline src/common/components/hosts/tableUtils.tsx:183]: translation

**Review by mareklibra — CHANGES_REQUESTED** [review]

File: locales/en/translation.json (hunk 2 above)
Comment (rawagner) [reply locales/en/translation.json:693]: CI should have catch that and fail. Can you take a look why it didnt ?

File: src/cim/components/featureSupportLevels/featureStateUtils.tsx (hunk 4 above)
Comment (jtomasek) [reply src/cim/components/featureSupportLevels/featureStateUtils.tsx:30]: The `clusterimageset` resource does not have any property which describes the image architecture. Only lead is the image name/url but that is very unreliable and fragile.

File: src/cim/components/ClusterDeployment/ArmCheckbox.tsx (hunk 1 above)
Comment (jtomasek) [reply src/cim/components/ClusterDeployment/ArmCheckbox.tsx]: done

File: locales/en/translation.json (hunk 2 above)
Comment (jtomasek) [reply locales/en/translation.json]: done and fixed the validation here: https://github.com/openshift-assisted/assisted-ui-lib/pull/1747/commits/a47343440d22873f5f0a93572b849b9f7698dde7

File: src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx (hunk 3 above)
Comment (jtomasek) [reply src/cim/components/ClusterDeployment/ClusterDeploymentHostsSelectionAdvanced.tsx]: done

File: src/common/components/hosts/tableUtils.tsx (hunk 5 above)
Comment (jtomasek) [reply src/common/components/hosts/tableUtils.tsx]: done

**Review by mareklibra — APPROVED** [review]

**Review by rawagner — DISMISSED** [review]

File: src/cim/components/helpers/clusterDeployment.ts [hunk 6]
```diff
@@ -199,3 +206,19 @@ export const getClusterProperties = (
...
+  infraEnv?: InfraEnvK8sResource,
+) => {
+  let arch;
+  if (
+    infraEnv &&
+    isEqual(infraEnv.spec?.clusterRef, pick(clusterDeployment.metadata, ['name', 'namespace']))
+  ) {
+    arch = infraEnv.spec?.cpuArchitecture;
+  }
+  arch = clusterDeployment.metadata?.annotations?.[CPU_ARCHITECTURE_ANNOTATION_KEY];
```
Comment (celdrake) [inline src/cim/components/helpers/clusterDeployment.ts:221]: Is it intentional? this overwrites the assignment in the IF

File: src/cim/components/ClusterDeployment/ArmCheckbox.tsx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+  HelperText,
+  PopoverIcon,
+  CpuArchitecture,
+  OpenshiftVersionOptionType,
+} from '../../../common';
+import {
+  FeatureSupportLevelBadge,
+  useFeatureSupportLevel,
+} from '../../../common/components/featureSupportLevels';
+import { useTranslation, useTranslation } from '../../../common/hooks/use-translation-wrapper';
```
Comment (celdrake) [inline src/cim/components/ClusterDeployment/ArmCheckbox.tsx:17]: There is TS error here
```suggestion
import { useTranslation } from '../../../common/hooks/use-translation-wrapper';
```

Comment (jtomasek) [reply src/cim/components/ClusterDeployment/ArmCheckbox.tsx]: Fixed

**Review by celdrake — APPROVED** [review]

**Review by rawagner — APPROVED** [review]

**Pull request merged**